### PR TITLE
MCA/ATOMICS/UCX: workaround for abstraction violation

### PR DESCRIPTION
**Pull request by hoopoepg**
- some spml calls are marked as inline to exclude cross-module
  dependency
- updated get-key call to get link to local module

workaround for issue #3224 which allows to load atomic module

Signed-off-by: Sergey Oblomov <sergeyo@mellanox.com>

### CODE REVIEW

**Comment by hoopoepg** [comment]
@yosefe what do you think, will it work?

File: oshmem/mca/spml/ucx/spml_ucx.h [hunk 1]
```diff
@@ -121,14 +121,28 @@ extern int mca_spml_ucx_quiet(void);
 extern int spml_ucx_progress(void);
 
 
-spml_ucx_mkey_t * mca_spml_ucx_get_mkey_slow(int pe, void *va, void **rva);
+static inline
```
[the remark this answers is not in the document]
Comment (yosefe) [reply oshmem/mca/spml/ucx/spml_ucx.h]: or maybe keep it in C file, but pass  `mca_spml_ucx_t* module` parameter to it

[the remark this answers is not in the document]
Comment (hoopoepg) [reply oshmem/mca/spml/ucx/spml_ucx.h]: inline call could affect to performance negatively?

ok, done

[the remark this answers is not in the document]
Comment (yosefe) [reply oshmem/mca/spml/ucx/spml_ucx.h]: yes, adding inline call which is rarely used uses more registers and bloats function prolog/epilog (push/pop)

File: oshmem/mca/spml/ucx/spml_ucx.h [hunk 2]
```diff
@@ -68,6 +68,7 @@ struct mca_spml_ucx {
 
     int                      priority; /* component priority */
     bool                     enabled;
+    spml_ucx_mkey_t          *(*mca_spml_ucx_get_mkey_slow)(int pe, void *va, void **rva);
```
Comment (yosefe) [inline oshmem/mca/spml/ucx/spml_ucx.h]: this could be renamed to just `get_mkey_slow`

Comment (hoopoepg) [reply oshmem/mca/spml/ucx/spml_ucx.h]: done

**Comment by ibm-ompi** [comment]
The IBM CI (PGI Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/2d6527864acf401e8ce56c067adc2043

**Comment by ibm-ompi** [comment]
The IBM CI (PGI Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/68a1f25fd4bc4a8acac0b1f31f25f3a9

File: oshmem/mca/spml/ucx/spml_ucx.c [hunk 3]
```diff
@@ -75,7 +78,9 @@ mca_spml_ucx_t mca_spml_ucx = {
     NULL,   /* ucp_peers */
     0,      /* using_mem_hooks */
     1,      /* num_disconnect */
-    0       /* heap_reg_nb */
+    0,      /* heap_reg_nb */
```
Comment (hoopoepg) [inline oshmem/mca/spml/ucx/spml_ucx.c]: @yosefe it seems here is something wrong with initialization: as I can see init parameters-comments doesn't correspond to mca_spml_ucx_t comments (mixed initialization values???)

**Comment by hoopoepg** [comment]
@yosefe ok to go?

**Pull request merged**